_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+name:  "Integrating rate limiting nullifiers into Waku Relay to enable spam-protection."
+title:  "Integrating rate limiting nullifiers into Waku Relay to enable spam-protection."
+date:   2021-02-25 12:00:00 +0800
+author: staheri14
+published: true
+permalink: /waku-v2-rln-relay
+categories: research
+summary: A simplified explanation of how Waku RLN-Relay mitigates the spamming issue in a p2p manner.
+image: 
+discuss: 
+---
+
+# Introduction
+
+This blog is going to give you an overview of how we enabled spam protection in waku relay through rate-limiting nullifiers or rln for short.
+
+Firstly, let me give a little background about Waku. Waku is a privacy-preserving peer-to-peer (p2p) messaging protocol for resource restricted devices. Being p2p means that Waku relies on **No** central server, instead messages get transmitted in the network by users' (i.e., peers) collaboration. The underlying protocol (as the writeup of this blog) is [GossipSub](https://docs.libp2p.io/concepts/publish-subscribe/). In a very informal way, the message transmission in GossipSub is like I send a message to you, you send it to whoever you know, and it goes on till the message reaches its destination. 
```

Comment:
The last sentence can probably be rewritten for clarity

Reply:
One thing I find useful is to do a pass of text through something like https://www.hemingwayapp.com/, that tends to make things like this clearer